src/discretization/regular.jl

```diff
@@ -57,7 +57,8 @@ function discretize(sphere::Sphere{3,T},
   # add north and south poles
   c = center(sphere)
   r = radius(sphere)
-  e⃗ = Vec{3,T}(0, 0, 1)
+  V = typeof(one(T))
```

Comment:
Could you please confirm that this is an alternative method for unit cancellation? https://painterqubits.github.io/Unitful.jl/stable/conversion/#Unit-cancellation

Reply:
Good question.

This is indeed a good way to eliminate units. But there are two problems with this approach.

1. This method can only be used to eliminate units that appear to be units for quantities that do not already have them. Which means m kN J^-1 equals to 1000, NoUnits just convert it to 1000. `1u"mm" |> NoUnits` will throw an error because millimeter could not convert to number. We want to drop the unit instead of convert it into numbers. We need a scaling factor instead of actual quantities.

2. `NoUnits` will make Unitful as our dependences which is not expected. Our package should be general. Not only Unitful but also other Unitful like packages should work well. Thus, we should not use any Unitful like packages as our dependences only approach it with `one`, `oneunit` and other methods which has already included by Base.

Reply:
Fully agree. Thank you for checking the nuances. There is also ustrip to "get the number in front" of the units but that would also make Unitful.jl a dependency. I think we can introduce a helper function in our utils.jl file if it already exists to drop units, something like `dropunits(T) = typeof(one(T))` and then use it consistently in places of the code where we find scaling factors. Can you adapt this PR accordingly?